Feature/organizations creation

Thomas-WG opened the pull request:
- Page to create, edit and delete organizations
- Function to load picture in the storage
- Function to create/edit/delete org in Firestore.

Thomas-WG committed:
Creation of the form, function to update firestore and reorg into components

Thomas-WG committed:
Add translation and send picture to storage

Thomas-WG committed:
Added translation and skills list dynamically from firestore

Thomas-WG committed:
Manage skills page

Thomas-WG committed:
added Create organization page + functions

Thomas-WG committed:
adding comment

opedoussaut commented:
**General Feedback**
Good Work! (so nice Mr copilot)
This PR introduces functionality for creating, editing, and deleting organizations, alongside uploading pictures to storage and managing Firestore entries. The implementation aligns well with the app's goals.

Specific Feedback by File

**firebase.json**
Comment:
The addition of storage emulator configurations is helpful. Ensure that the import/export paths for .firebase/emulator-data are correctly set and tested during development.
opt2: probably something to check for future releases of the website not for this preliminary version.

**src/app/login/page.js**
Comment:
The expanded user profile initialization is a welcome addition. Consider adding validation for critical fields like time Commitment and availabilities to ensure consistency and prevent errors.
opt2: I didn't check the validation steps you have added to the various fields, maybe we should discuss content validation versus WG policy for those fields ? 

**package.json**
Comment:
Dependencies such as @cospired/i18n-iso-languages and countries-list are useful for localization and country management. Are these fully utilized, or could we find lighter alternatives to reduce overhead?
opt2: I might have some proposal for lighter alternatives, we'll discuss later on about this. 

**messages/*.json**
Comment:
The new localization strings for organization management are extensive. However, I suggest including placeholders or defaults for any missing translations to prevent fallback issues. For instance:
Use a default locale (en.json) as the base for all other translations. Any missing keys in other locales (e.g., es.json or fr.json) could display placeholders like [MISSING TRANSLATION] or fall back to the English text.
Example:
If editOrganization is missing in es.json, it could have a placeholder value:
JSON
"editOrganization": "[TRADUCCIÓN FALTANTE]"
Alternatively, configure a localization tool/library to automatically fall back to the default language (e.g., English) for missing keys:
JavaScript
i18n.configure({
  locales: ['en', 'es'],
  defaultLocale: 'en',
  missingKeyFn: (locale, value) => `[MISSING TRANSLATION: ${value}]`
});
This approach ensures that the user interface remains consistent and prevents broken text or empty content when translations are incomplete.
opt2: 
I'll have a better solution I think for translation but we'll discuss about it later on, I found this proposal interesting in case we don't want to spend too much time first on translation but I think you have this covered already...

**Next Steps**
Testing:
Test the organization management workflow thoroughly across all supported locales and configurations.
Documentation:
Update any related documentation or README sections to reflect the new features and added dependencies.
Validation:
Add validation or constraints for fields like languages, skills, and availabilities to ensure data integrity.

Thomas-WG commented:
Thanks for the feedbacks ! A few comments/answers:
#

> Specific Feedback by File
> 
> **firebase.json** Comment: The addition of storage emulator configurations is helpful. Ensure that the import/export paths for .firebase/emulator-data are correctly set and tested during development. opt2: probably something to check for future releases of the website not for this preliminary version.
==> thomas: Completely agreed, I added it as it will save some time during the dev.

> **src/app/login/page.js** Comment: The expanded user profile initialization is a welcome addition. Consider adding validation for critical fields like time Commitment and availabilities to ensure consistency and prevent errors. opt2: I didn't check the validation steps you have added to the various fields, maybe we should discuss content validation versus WG policy for those fields ?
==> thomas: I haven’t added any check on Commitment and availabilities as it’s not critical. I do check at minimum Display name, country and languages.

> **package.json** Comment: Dependencies such as @cospired/i18n-iso-languages and countries-list are useful for localization and country management. Are these fully utilized, or could we find lighter alternatives to reduce overhead? opt2: I might have some proposal for lighter alternatives, we'll discuss later on about this.
==> Thomas: sure, I’m opened. I was lazy on that one and just used an existing library with the full list. At least it is aligned between the Organization Country&Languages, and the Member’s one.
> 
> **messages/*.json** Comment: The new localization strings for organization management are extensive. However, I suggest including placeholders or defaults for any missing translations to prevent fallback issues. For instance: Use a default locale (en.json) as the base for all other translations. Any missing keys in other locales (e.g., es.json or fr.json) could display placeholders like [MISSING TRANSLATION] or fall back to the English text. Example: If editOrganization is missing in es.json, it could have a placeholder value: JSON "editOrganization": "[TRADUCCIÓN FALTANTE]" Alternatively, configure a localization tool/library to automatically fall back to the default language (e.g., English) for missing keys: JavaScript i18n.configure({ locales: ['en', 'es'], defaultLocale: 'en', missingKeyFn: (locale, value) => `[MISSING TRANSLATION: ${value}]` }); This approach ensures that the user interface remains consistent and prevents broken text or empty content when translations are incomplete. opt2: I'll have a better solution I think for translation but we'll discuss about it later on, I found this proposal interesting in case we don't want to spend too much time first on translation but I think you have this covered already...
==> Thomas: Indeed that might be good to see for a placeholder (I’m not sure how much rework it will cost) but Cursor manage the translations of the different locales.
> 
> **Next Steps** Testing: Test the organization management workflow thoroughly across all supported locales and configurations. Documentation: Update any related documentation or README sections to reflect the new features and added dependencies. Validation: Add validation or constraints for fields like languages, skills, and availabilities to ensure data integrity.
